src/main/java/org/springframework/data/couchbase/repository/query/CouchbaseQueryMethod.java

```diff
@@ -163,6 +178,26 @@ public String getInlineN1qlQuery() {
 		return StringUtils.hasText(query) ? query : null;
 	}
 
+	/**
+	 * is this a 'delete'?
+	 *
+	 * @return is this a 'delete'?
+	 */
+	public boolean isDeleteQuery() {
+		return getName().toLowerCase().startsWith("delete");
```

Comment:
Is spring-core guaranteed to be on the class path? Maybe you can use [StringUtils.startsWithIgnoreCase](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/util/StringUtils.html#startsWithIgnoreCase-java.lang.String-java.lang.String-)(getName(), "delete");